astropy_helpers/tests/__init__.py

```diff
@@ -39,18 +37,24 @@ def run_cmd(cmd, args, path=None, raise_error=True):
     return streams + (return_code,)
 
 
-@extends_doc(sandbox.run_setup)
-def run_setup(*args, **kwargs):
-    """
-    In Python 3, on MacOS X, the import cache has to be invalidated otherwise
-    new extensions built with ``run_setup`` do not always get picked up.
-    """
+def run_setup(setup_script, args):
+
+    # This used to call setuptools.sandbox's run_setup, but due to issues with
+    # this and Cython (which caused segmentation faults), we now use
+    # subprocess.
+
+    path = os.path.dirname(setup_script)
+    setup_script = os.path.basename(setup_script)
+
+    if not path:
+        path = None
 
-    try:
-        return sandbox.run_setup(*args, **kwargs)
-    finally:
-        import importlib
-        importlib.invalidate_caches()
+    p = sp.Popen([sys.executable, setup_script] + list(args), cwd=path,
+                 stdout=sp.PIPE, stderr=sp.PIPE)
+    stdout, stderr = p.communicate()
+
+    sys.stdout.write(stdout.decode('utf-8'))
+    sys.stderr.write(stderr.decode('utf-8'))
```

Comment:
Isn't this subtly different in that the process has to *finish* before any output is visible?  So any time this is used the process will appear to hang?  Why not instead pass stdout/stderr in to the Popen?

Reply:
Just to clarify, this ``run_setup`` is used **only** in the tests, and the only ``run_setup`` call in ``ah_bootstrap.py`` was in a ``silence`` context manager (it was for the helpers, not the actual package being built). The ``run_setup`` function is not used in the astropy build process.